library/src/main/java/com/okta/appauth/android/OAuthClientConfiguration.java

```diff
@@ -193,10 +194,13 @@ void readConfiguration(@NonNull final JSONObject jsonObject)
             throws InvalidJsonDocumentException {
         JsonParser jsonParser = JsonParser.forJson(jsonObject);
 
+        //We can not take has code directly from JSONObject
+        //because JSONObject does not follow java has code contract
         mConfigHash = jsonObject.toString().hashCode();
 
         mClientId = jsonParser.getRequiredString("client_id");
         mRedirectUri = jsonParser.getRequiredUri("redirect_uri");
+        mEndSessionRedirectUri = jsonParser.getRequiredUri("end_session_redirect_uri");
```

Comment:
Below, we assert that the redirect URI is registered. Any reason we chose not to do so with the end session redirect URI?

Reply:
As long as we have one URI schema for redirectUri and endSessionUri we don't have to check for registration of both URI's in application manifest, but I will do renaming of the `isRedirectUriRegistered` method as from now it is checking URI for endSession as well as for redirect.